src/SBInterpolatedImage.cpp

```diff
@@ -895,31 +881,61 @@ namespace galsim {
         }
         _flux = kValue(Position<double>(0.,0.)).real();
         dbg<<"flux = "<<_flux<<std::endl;
+        setCentroid();
+    }
+
+    void SBInterpolatedKImage::SBInterpolatedKImageImpl::setCentroid() const {
```

Comment:
OK.  I guess this might be premature optimization.  I don't know that the centroid calculation is a tall pole of optimization for any use case with this class.  If it turns out that this is a slow step, it's not too hard to bring back the delayed calculation.